pkg/test/builder/taskrun.go

```diff
@@ -14,3 +15,23 @@ func TaskRunCompletionTime(ct time.Time) tb.TaskRunStatusOp {
 		s.CompletionTime = &metav1.Time{Time: ct}
 	}
 }
+
+// StepState adds a StepState to the TaskRunStatus.
+func StepState(ops ...tb.StepStateOp) tb.TaskRunStatusOp {
+	return func(s *v1alpha1.TaskRunStatus) {
+		state := &v1alpha1.StepState{}
+		for _, op := range ops {
+			op(state)
+			s.Steps = append(s.Steps, *state)
```

Comment:
This should  be out of the loop — you "build" the state, then you add it.

Reply:
good catch :+1:  , resolved it  :sweat_smile: